src/lexer/lexer.cc

```diff
@@ -0,0 +1,35 @@
+#include "lexer/lexer.h"
+
+#include <fstream>
+#include <sstream>
+
+namespace lexer {
+
+namespace {
+
+std::istream* get_stream_from_source(const std::string& source, Lexer::SourceTag tag) {
+  switch (tag) {
+    case Lexer::SourceTag::FILE:
+      return new std::ifstream(source);
+    case Lexer::SourceTag::STRING:
+      return new std::stringstream(source);
+    default:
+      return nullptr;
+  }
+}
+
+} // namespace
+
+Lexer::Lexer(const std::string& source, SourceTag tag)
+  : stream_{get_stream_from_source(source, tag)} {}
+
+Token Lexer::get_next_token() {
+  Token result;
+  result.type = TokenType::END_OF_FILE;
+  result.begin.line = line_;
+  result.begin.column = column_;
+  result.end = result.begin;
+  return result;
+}
+
+} // lexer
```

Comment:
Same as above about spacing of comments. Also, you need "// namespace lexer" and not only "// lexer"